codalab/model/bundle_model.py

```diff
@@ -1001,12 +1001,16 @@ def transition_bundle_running(self, bundle, worker_run, row, user_id, worker_id,
             'remote': worker_run.remote,
             'cpu_usage': cpu_usage,
             'memory_usage': memory_usage,
+            'data_size': worker_run.disk_utilization,
         }
 
-        # Increment user time as we go to ensure user doesn't go over time quota.
+        # Increment user time and disk as we go to ensure user doesn't go over quota.
         if user_id == self.root_user_id and hasattr(bundle.metadata, 'time'):
             time_increment = worker_run.container_time_total - bundle.metadata.time
             self.increment_user_time_used(bundle.owner_id, time_increment)
+        if hasattr(bundle.metadata, 'data_size'):
+            disk_increment = worker_run.disk_utilization - bundle.metadata.data_size
+            self.increment_user_disk_used(bundle.owner_id, disk_increment)
```

Comment:
Can you explain a bit... where was `increment_user_disk_used` called previously? and are we still calling it in the previously-called place?

Reply:
Sure thing!

So, even after this change, the user disk quota is updated [in this function](https://github.com/codalab/codalab-worksheets/blob/master/codalab/model/bundle_model.py#L1101) [here](https://github.com/codalab/codalab-worksheets/blob/master/codalab/model/bundle_model.py#L1118) upon a bundle finishing. This actually recomputes the user disk quota entirely from scratch (see the function call [here](https://github.com/codalab/codalab-worksheets/blob/94b5ca0dd17d195decea847669e70a38e806bc49/codalab/model/bundle_model.py#L1156) and its definition [here](https://github.com/codalab/codalab-worksheets/blob/94b5ca0dd17d195decea847669e70a38e806bc49/codalab/model/bundle_model.py#L2776)).

Basically, then, increment_user_disk_used isn't currently being called anywhere except in bundles/rest.py; I actually added that function for a separate change I made last quarter. I use it here to just increment the user disk as the bundle runs, but the user disk quota update is done the exact same as before after the bundle finishes.

I think, overall, that the way the disk quota currently is updated and is managed is a bit confusing and oftentimes slows things down, and I actually think we could really speed up CodaLab by making our handling of the disk quota a bit tighter.